File: BackEnd/Models/Videolist.cs
using System;
using System.Collections.Generic;

namespace Showreel.Models;

public partial class Videolist
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string? Remark { get; set; }

    public DateTime? CreatedTime { get; set; }

    public DateTime? LastUpdatedTime { get; set; }

    public virtual ICollection<VideoVideolist> VideoVideolists { get; set; } = new List<VideoVideolist>();
}

File: FrontEnd/src/model/videoList.ts
import { Building } from './building';
import { Category } from './category';
import { Video } from './video';
import { VideoVideolist } from './videoVideolist';

export interface VideoList {
    id?: number;
    title?: string;
    remark?: string;
    createdTime?: Date;
    lastUpdatedTime?: Date;
    videoVideoList?: VideoVideolist[];
}

export interface IVideoInList {
    video: Video;
    category: Category[];
    loopNum: number;
    noBackToBack?: Category[];
    doNotPlay?: Building[];
}
